registry/storage/manifeststore.go

```diff
@@ -13,6 +15,149 @@ import (
 	"github.com/docker/distribution/manifest/schema2"
 )
 
+const (
+	// MediaTypeOrphanedManifestRevision specifies the mediaType for orphaned
+	// (unlinked, delted) manifest that can be cleaned up. This type of
+	// manifest can be returned only during enumeration if enabled with option.
+	MediaTypeOrphanedManifestRevision = "application/vnd.docker.distribution.manifest.orphaned.v2+json"
```

Comment:
If I understand this correctly, this is to handle the case where the manifest revision is part of the repository but the data is no longer available?

Going back the the enumerate option would probably be okay. We can have a help function that does the manifest lookup:

```
func LookupManifest(service ManifestService, fn func(id digest.Digest, manifest Manifest, err error)) func(digest.Digest) error
```

If an error is encountered, `fn` can be called with an id, nil manifest and error, and the case of missing manifests can be solved!